### PR TITLE
Add exception details to error logging for blob repository

### PR DESCRIPTION
**Pull request by liliankasem**
Logging the exception for blob repository errors so we have more details when a failure occurs. 

Related to #7591

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [x] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

**before**
Screen Shot 2022-03-29 at 6 56 09 PM

**after**
Screen Shot 2022-03-30 at 4 56 11 PM

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs [hunk 1]
```diff
@@ -14,22 +14,22 @@ public static class ScriptHostServiceSecurityExtension
             LoggerMessage.Define<string, string>(
                 LogLevel.Error,
                 new EventId(600, nameof(BlobStorageSecretRepoError)),
-                "There was an error performing a {operation} operation on the Blob Storage Secret Repository. Please ensure the '{appSettingName}' connection string is valid.");
+                "There was an error performing a {operation} operation on the Blob Storage Secret Repository. Please ensure the '{appSettingName}' connection string is valid. {exception}.");
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: Do we need to explicitly include the exception in the format string? I think [it gets passed to the internal Log method](https://github.com/dotnet/runtime/blob/37c5a3e6bf685bb08550baba54a742d21c91ffe7/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerMessage.cs#L184) automagically.

Comment (kshyju) [reply src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: Also, the number of named parameters in message string (3) not matching with the definition (2).

Comment (liliankasem) [reply src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: From the kusto logs I was looking at, we don't see any exception details

Comment (liliankasem) [reply src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: @kshyju I doubled checked this and you're right it is included. So I will remove it from the message string :)

File: src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs [hunk 2]
```diff
@@ -13,13 +13,13 @@ public static class ScriptHostServiceSecurityExtension
             LoggerMessage.Define<string, string>(
                 LogLevel.Error,
                 new EventId(600, nameof(BlobStorageSecretRepoError)),
-                "There was an error performing a {operation} operation on the Blob Storage Secret Repository.");
+                "There was an error performing a {operation} operation on the Blob Storage Secret Repository. Please ensure the '{appSettingName}' connection string is valid.");
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: I wonder the "_Please ensure the '{appSettingName}' connection string is valid_." part of message may cause confusion. Is it possible that this error logging code is executed for cases where the connection string is valid, but the READ/WRITE/LIST operations threw an exception for other reasons? For example, the `BlobContainerClient.GetBlobsAsync` could [throw a RequestFailedException](https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.blobcontainerclient.getblobsasync?view=azure-dotnet#remarks)

**Review by kshyju — APPROVED** [review]
LGTM. One question about the message we are logging.

File: src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs (hunk 2 above)
Comment (liliankasem) [reply src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceSecurityExtension.cs]: Yeah I am debating this one myself, it's almost a red herring and is might be best removed. But yes, this error shows up when it is not an auth issue which is where this PR came from in the first place :D 

On one hand, it's useful information when the issue is auth related, on the other hand if the issue is something else, this is useless information. I think I'm leaning towards removing that part of the string and just making sure our documentation is updated. 

Something else worth noting, I am not seeing this error surface in AI or the log stream - should it be? Or should we leave this for our internal logs?

**Review by satvu — APPROVED** [review]

**Pull request merged**